### ISSUE TITLE
Edge does not actually support SpeechRecognition Web Speech API

### ISSUE THREAD

**Issue by mbforbes** [mdn/browser-compat-data#22126]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/SpeechRecognition

### What specific section or headline is this issue about?

Browser compatibility

### What information was incorrect, unhelpful, or incomplete?

Microsoft Edge does not actually support the SpeechRecognition Web Speech API. It's listed that it is "Implemented with the vendor prefix: webkit." While using it does not generate any errors, the implementation does nothing. No results are ever returned.

This may be an issue of semantics. Does implementing with a no-op count as implementing? :-)

### What did you expect to see?

The Edge column should either have a "No" or a warning sign because the implementation is broken.

### Do you have any supporting links, references, or citations?

Two users have tested on two different devices, each on two separate pages (4 total tests). For all of them, Chrome's speech recognition works correctly, and Edge never returns any results.

n=4 isn't huge, but it's been consistent. I also believe this issue is known by Edge.

Here's an example test page this can be used to verify SpeechRecognition on the Web Speech API:
https://www.google.com/intl/en/chrome/demos/speech.html

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/speechrecognition`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/SpeechRecognition
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/speechrecognition/index.md
* Last commit: https://github.com/mdn/content/commit/6b1e3eebf22abf1b73bb219581335b1147b75d7a
* Document last modified: 2023-03-12T09:29:44.000Z

</details>

**Comment by bsmth** [comment]
Hi @mbforbes thanks a lot for raising this. I've moved it over to the browser-compat-data repo which is where the source of this compatibility data is managed.

Just to check, there are some demos to try to reproduce this. I did the following:


```bash
git clone https://github.com/mdn/dom-examples.git
cd dom-examples/web-speech-api/speech-color-changer
http-server -c-1
# Available on:
#   http://127.0.0.1:8080
```

This works on Chrome, but fails on Edge with "Language not supported" error. You may want to try the other examples in that repo, but that doesn't look promising to me.

See https://github.com/mdn/dom-examples/blob/40cb0f10d859384a6160861ccbbf78f337fed6a0/web-speech-api/speech-color-changer/script.js#L17

**Comment by mbforbes** [comment]
Hey @bsmth thank you for the reply, and apologies for my newb raising of this issue in the wrong repo 😅 I appreciate you looking into it already!

**Comment by bsmth** [comment]
> Hey @bsmth thank you for the reply, and apologies for my newb raising of this issue in the wrong repo 😅 I appreciate you looking into it already!

Not a problem, I hope the info helps investigate / narrow it down.

**Comment by gabrieljbaker** [comment]
@mbforbes , I've been trying to get recognition to work in Edge for a while - definitely confirmed that it does not work.  Wild that Safari has this and Edge doesn't.

**Comment by mbforbes** [comment]
So, with a higher sample size, the plot has thickened. I have confirmed reports of Edge working for _some_ folks. But for others, it doesn't work at all. Anecdotally, it's about 1:4 working vs not, but the _n_ is too small. No idea when vs not.

**Comment by bsmth** [comment]
> So, with a higher sample size, the plot has thickened. I have confirmed reports of Edge working for _some_ folks. But for others, it doesn't work at all. Anecdotally, it's about 1:4 working vs not, but the _n_ is too small. No idea when vs not.

For those that report it working, could you get some information on operating system and browser version? That would really help figure out what's happening here.

**Comment by mobilepadawan** [comment]
Does anyone know if there is exists an estimated date to start supporting Edge on Mac OS over ARM processor?

Chrome is working fine in this O.S. and processor.

**Comment by bsmth** [comment]
>here's some color to it

Thanks for picking that up. Here's a link to the post in question for convenience: https://twitter.com/khmyznikov/status/1772573147755073667

I think we could add a note for Edge entry that this depends on Azure Speech component & is not yet supported on ARM machines, it would be great to link to a bug tracker or some other official docs for this.

**Comment by sushraja-msft** [comment]
@caugner - I am assuming this macOS machine is an ARM mac device? We have a bug in Edge for ARM devices where the component is not being delivered. Ill update when that is fixed.

**Comment by caugner** [comment]
> I am assuming this macOS machine is an ARM mac device?

@sushraja-msft Good catch: I'm on a Macbook M1 Pro myself, and in BrowserStack Live, I can reproduce the issue in Edge 136 on macOS, but not in Edge 136 on Windows 11.

Will this only be fixed in future versions, or will Edge 136 and older versions be also fixed retroactively?

**Comment by gabrieljbaker** [comment]
any update on that @sushraja-msft ?

**Comment by sushraja-msft** [comment]
We are working on a rewrite with better proxy and ARM support. I don't have an ETA yet, still in prototyping stages, but this is being worked on.

**Comment by mobilepadawan** [comment]
**My apologize if my next comment sounds like an ape.**

I have no knowledge at all about if there is exists a Chromium for MacOS (_a blank branding of the blink engine_) with this feature working on it.  I suppose could be a good choice to take that functional part of this feature to accelerate the support on Edge.

I teach advanced programming skills with JS in a tertiary school, and also I have in mind a couple of web projects where this feature could be a highlight. 

Every time I should teach Voice Recognition in JS I need to switch to Chrome.

**Comment by sushraja-msft** [comment]
The implementation in Chromium uses Google's speech service. So, this is something other Chromium browsers have to re-implement unfortunately.
